cd /home/pt40963/scratch/pea_aphid_RNAseq_20171104

#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/avr-rep1Aligned.sortedByCoord.out.bam_sorted.bam avr-rep1Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/N116-rep3Aligned.sortedByCoord.out.bam_sorted.bam N116-rep3Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/PSO1-rep5Aligned.sortedByCoord.out.bam_sorted.bam PSO1-rep5Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/avr-rep2Aligned.sortedByCoord.out.bam_sorted.bam avr-rep2Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/N116-rep4Aligned.sortedByCoord.out.bam_sorted.bam N116-rep4Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/vir-rep1Aligned.sortedByCoord.out.bam_sorted.bam vir-rep1Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/avr-rep3Aligned.sortedByCoord.out.bam_sorted.bam avr-rep3Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/N116-rep5Aligned.sortedByCoord.out.bam_sorted.bam N116-rep5Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/vir-rep2Aligned.sortedByCoord.out.bam_sorted.bam vir-rep2Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/avr-rep4Aligned.sortedByCoord.out.bam_sorted.bam avr-rep4Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/PSO1-rep1Aligned.sortedByCoord.out.bam_sorted.bam PSO1-rep1Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/vir-rep3Aligned.sortedByCoord.out.bam_sorted.bam vir-rep3Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/avr-rep5Aligned.sortedByCoord.out.bam_sorted.bam avr-rep5Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/PSO1-rep2Aligned.sortedByCoord.out.bam_sorted.bam PSO1-rep2Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/vir-rep5Aligned.sortedByCoord.out.bam_sorted.bam vir-rep5Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/N116-rep1Aligned.sortedByCoord.out.bam_sorted.bam N116-rep1Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/PSO1-rep3Aligned.sortedByCoord.out.bam_sorted.bam PSO1-rep3Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/N116-rep2Aligned.sortedByCoord.out.bam_sorted.bam N116-rep2Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts
#
#python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name /home/pt40963/scratch/pea_aphid_RNAseq_20171104/shell_scripts/de_exons/A.pisum_exons.gff /home/pt40963/scratch/pea_aphid_RNAseq_20171104/PSO1-rep4Aligned.sortedByCoord.out.bam_sorted.bam PSO1-rep4Aligned.sortedByCoord.out.bam_sorted.bam.exons.counts



#############################################################################################################################################################
# Title: work flow to find differential exon expression

# REAL EXMAPLE AT THE END!!!

#required software:
#
#STAR		 	http://labshare.cshl.edu/shares/gingeraslab/www-data/dobin/STAR/STAR.posix/doc/STARmanual.pdf
#Python 2.7
#HTseq			http://www-huber.embl.de/HTSeq/doc/install.html#install
#Pysam			https://github.com/pysam-developers/pysam
#DEXseq    		https://bioconductor.org/packages/3.3/bioc/vignettes/DEXSeq/inst/doc/DEXSeq.pdf
#R
#####################################################################################################################################################################################
#
#1) Map with spliced aware aligner:   MAP to genome using STAR     
#
#
#mkdir star_indicies
#
## index the genome - up the RAM linit to 70GB
#$ STAR  --runMode genomeGenerate --runThreadN 10 --limitGenomeGenerateRAM 74554136874 --genomeDir /home/pt40963/M.cerasi_genome_data/final_assembly_v1/star_indicies --genomeFastaFiles /home/pt40963/M.cerasi_genome_data/final_assembly_v1/Myzus_cerasi_genome_assembly.v1.fasta
#
## possible new command: map the RNAseq  slightly stricter settings than default. OUTPUT as sorted BAM if pysam installed for later scripts.
#
#$ STAR --genomeDir star_indicies/  --runThreadN 10 
#--outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 5 
#--outFilterMismatchNoverLmax 0.2 --readFilesCommand zcat 
#--readFilesIn /home/pt40963/M.cerasi_genome_data/RNAseq_data/Mc_R1.fq.gz 
#/home/pt40963/M.cerasi_genome_data/RNAseq_data/Mc_R2.fq.gz --outFileNamePrefix Mc_RNAseq_mapped
# 
#
#
#$ STAR --genomeDir star_indicies/  --runThreadN 10 --outSAMtype BAM --outFilterMismatchNmax 5
#--readFilesCommand zcat --readFilesIn /home/pt40963/Pea_aphid/RNAseq/all_r1_paired.fq.gz 
#/home/pt40963/Pea_aphid/RNAseq/all_r2_paired.fq.gz --outFileNamePrefix Pea_RANseq_mapped
#--outReadsUnmapped fastq     (this will output reads that do not map. Vry useful)
#
#
#####################################################################################################################################################################################
#
#2) prepare EXON bins from annotation.
#use the following .py script, use genome GTF (recommended), or possible GFF
#
#$ dexseq_prepare_annotation.py genome.gtf exons.gff
#
#		gt gff3 -sort -tidy -sortlines -checkids -fixregionboundaries -force -o GTOOLS2.gff Rpa.v1_alt.gff
#
#		gt gff3_to_gtf -force -o GTOOLS.gtf GTOOLS2.gff 
#		
#		python /home/pt40963/scratch/dexseq_prepare_annotation.py GTOOLS.gtf Rpadi_DEXSEQ.gff
#
#
#3) count reads, repeat for each experimental condition:
#
#$ dexseq_count.py -p yes -f bam exons.gff RNAseq_condition1.bam condition1.txt   (-p pair end reads).
#
#		python /home/pt40963/scratch/dexseq_count.py -p yes -f bam -r name Rpadi_DEXSEQ.gff Rp_art_24h_1Aligned.toTranscriptome.out.bam Rp_art_24h_1Aligned.toTranscriptome.out.exon.counts
#		repeat!!!
#
#
#####################################################################################################################################################################################
## RUN IN ONE COMMAND:
#
#simplest  case,  construct  the DEXSeqDataSet as  shown  in  Section  2  or  in  Appendix  B (ALL OF SECTION 4 in this document),  then
#run DEXSeq passing the DEXSeqDataSet as only argument, this function will output a DEXSeqResults
#object.
#
#dxr = DEXSeq(dxd)
#
#class(dxr)
#
#####################################################################################################################################################################################
#
#
#
#4) Read the data into R:
#
#R
##inDir = system.file("extdata", package="pasilla")
#
#basically this line is just for the example.
#
#type getwd()
#
#inDir = "/home/blah/bleh"
#
#countFiles = list.files(inDir, pattern="fb.txt$", full.names=TRUE)
#
#basename(countFiles)
#
#flattenedFile = list.files(inDir, pattern="gff$", full.names=TRUE)
#
#basename(flattenedFile)
#
#
## [1] "treated1fb.txt"   "treated2fb.txt"   "treated3fb.txt"   "untreated1fb.txt"
### [5] "untreated2fb.txt" "untreated3fb.txt" "untreated4fb.txt"
#
#sampleTable = data.frame(row.names = c( "treated1", "treated2", "treated3","untreated1", "untreated2", "untreated3", "untreated4" ),\
#				condition = c("knockdown", "knockdown", "knockdown","control", "control", "control", "control" ),\
#				libType = c( "single-end", "paired-end", "paired-end","single-end", "single-end", "paired-end", "paired-end"))
#
#
#or prepare a .csv file and R function read.csv it
#
#
#sampleTable
###            condition    libType
### treated1   knockdown single-end
### treated2   knockdown paired-end
### treated3   knockdown paired-end
### untreated1   control single-end
### untreated2   control single-end
### untreated3   control paired-end
### untreated4   control paired-end
#
#
#suppresssPackageStartupMessages( library( "DEXSeq" ) )
#
#
#dxd = DEXSeqDataSetFromHTSeq(countFiles,sampleData=sampleTable,design= ~ sample + exon + condition:exon,flattenedfile=flattenedFile )
#
## create a file with gene name, grep genome.gff | cut -f gene_name > geneIDsinsubset.txt
#
#
#genesForSubset = read.table(file.path(inDir, "geneIDsinsubset.txt"),stringsAsFactors=FALSE)[[1]]
#
#
# dxd = dxd[geneIDs( dxd ) %in% genesForSubset,]
#
#colData(dxd)
#
#
#
#head( counts(dxd), 5)
#
# 
#head( featureCounts(dxd), 5) 
#
# head ( rowRanges(dxd), 3 )
#
#sampleAnnotation( dxd )
#
#
#
#####################################################################################################################################################################################
#4) normalisation
#
## can use multiple core if using  package BiocParallel,  and  implemented  the BPPARAM
#
##multi core doesnt work with my curretn set up
#BPPARAM = MultiCoreParam(workers=4)
#
#
## uses DEseq2 method
#dxd = estimateSizeFactors( dxd )
#
#
#
## dispersion estimation
#dxd = estimateDispersions( dxd)
#
#		# multi core
#		dxd = estimateDispersions( dxd, BPPARAM=BPPARAM)
#
#
#
#plotDispEsts( dxd )
#
#
## test for DE (exon) expression) 
#dxd = testForDEU( dxd )
#
#		# multi core
#		dxd = testForDEU( dxd, BPPARAM=BPPARAM)
#
#
#
## estimate fold change
#
#dxd = estimateExonFoldChanges( dxd, fitExpToVar="condition")
#
#
#dxr1 = DEXSeqResults( dxd )
#
#dxr1
#
##From this object, we can ask how many genes are signicant with a false discovery rate of 5%:
##table ( dxr1$padj < 0.05 ) - too slack!!
#
#table ( dxr1$padj < 0.01 )
#
## ask how many genes are aected
#table ( tapply( dxr1$padj < 0.1, dxr1$groupID, any ) )
#
#
#plotMA(dxrq, cex=0.8)
#
#
#
#####################################################################################################################################################################################
#4) visualization
## put in the appropriate gene name of interest!!!!!!!
#
#plotDEXSeq( dxr1, "FBgn0010909", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
#
##another
#
## put in the appropriate gene name of interest!!!!!!!
#
#plotDEXSeq( dxr1, "FBgn0010909", displayTranscripts=TRUE, legend=TRUE,cex.axis=1.2, cex=1.3, lwd=2 )
#
#
## to display it with the normalised counts.
#
#
#plotDEXSeq( dxr1, "FBgn0010909", expression=FALSE, norCounts=TRUE,legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
#
#
#plotDEXSeq( dxr1, "FBgn0010909", expression=FALSE, splicing=TRUE,legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
#
#
## WRITE TO BROWSEABLE HTML USING DEXSeqHTML
#
#
#DEXSeqHTML( dxr1, FDR=0.01, color=c("#FF000080", "#0000FF80") 
#
#
#DEXSeqHTML( dxr1, FDR=0.01) # colours dont work
#
#
#
############################################################################################################################################################################
##################################################################################################################################################
#R
##inDir = system.file("extdata", package="pasilla")
#
#
#inDir = '/home/peter/Desktop/exon_DE'
#
#countFiles = list.files(inDir, pattern=".exon_counts$", full.names=TRUE)
#
#basename(countFiles)
#
#flattenedFile = list.files(inDir, pattern="gff$", full.names=TRUE)
#
#basename(flattenedFile)
#
#
## [1] "treated1fb.txt"   "treated2fb.txt"   "treated3fb.txt"   "untreated1fb.txt"
### [5] "untreated2fb.txt" "untreated3fb.txt" "untreated4fb.txt"
#
#				
#				
#				
#sampleTable = data.frame(row.names = c("N116_A17_1", "N116_A17_2", "N116_A17_3", "N116_D2A_1", "N116_D2A_2", "N116_D2A_3", "PS01_A17_1", "PS01_A17_2", "PS01_A17_3", "PS01_D2A_1", "PS01_D2A_2", "PS01_D2A_3"), condition = c("N116_A17", "N116_A17", "N116_A17", "N116_D2A", "N116_D2A", "N116_D2A", "PS01_A17", "PS01_A17", "PS01_A17", "PS01_D2A", "PS01_D2A", "PS01_D2A"),libType = c("paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end", "paired-end"))
#				
#
#sampleTable
#
#
##dxd =read.HTSeqCounts(countFiles,sampleData=sampleTable,design= ~ sample + exon + condition:exon,flattenedfile=flattenedFile) 
#
#
#library("DEXSeq")
#
#
#######################################################################################################
#
#
#dxd = DEXSeqDataSetFromHTSeq(countFiles,sampleData=sampleTable,design= ~ sample + exon + condition:exon,flattenedfile=flattenedFile)
#
## create a file with gene name, grep genome.gff | cut -f gene_name > geneIDsinsubset.txt
#
#
#genesForSubset = read.table(file.path(inDir, "geneIDsinsubset.txt"),stringsAsFactors=FALSE)[[1])
#
#
#dxd = dxd[geneIDs( dxd ) %in% genesForSubset,]
#
#colData(dxd)
#
#
#
#head( counts(dxd), 5)
#
# 
#head( featureCounts(dxd), 5) 
#
# head ( rowRanges(dxd), 3 )
#
#sampleAnnotation( dxd )
#
#
#
#####################################################################################################################################################################################
#4) normalisation
#
## can use multiple core if using  package BiocParallel,  and  implemented  the BPPARAM
#
#
##BPPARAM = MultiCoreParam(workers=3)
#
#
## uses DEseq2 method
#dxd = estimateSizeFactors( dxd )
#
#
#
## dispersion estimation
#dxd = estimateDispersions( dxd)
#
#		# multi core
#		#dxd = estimateDispersions( dxd, BPPARAM=BPPARAM)
#
#
#
#plotDispEsts( dxd )
#
#
## test for DE (exon) expression) 
#dxd = testForDEU( dxd )
#
#		# multi core
#		dxd = testForDEU( dxd, BPPARAM=BPPARAM)
#
#
#
## estimate fold change
#
#dxd = estimateExonFoldChanges( dxd, fitExpToVar="condition")
#
#
#dxr1 = DEXSeqResults( dxd )
#
#write.table(dxr1, file = "Dexseq_results.out", append = FALSE, quote = TRUE, sep = "\t",
#            eol = "\n", na = "NA", dec = ".", row.names = TRUE,
#            col.names = TRUE, qmethod = c("escape", "double"),
#            fileEncoding = "")
#
#dxr1
#
##From this object, we can ask how many genes are signicant with a false discovery rate of 5%:
#table ( dxr1$padj < 0.05 )
#
##From this object, we can ask how many genes are signicant with a false discovery rate of 1%:
#table ( dxr1$padj < 0.01 )
#
## ask how many genes are aected (FDR 0.001)
#table ( tapply( dxr1$padj < 0.001, dxr1$groupID, any ) )
#
#
#plotMA(dxr1, cex=0.8)
#
#
#
#####################################################################################################################################################################################
#4) visualization
## put in the appropriate gene name of interest!!!!!!!
#
##plotDEXSeq( dxr1, "FBgn0010909", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
#plotDEXSeq( dxr1, "1005", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
##another
#
## put in the appropriate gene name of interest!!!!!!!
#
##plotDEXSeq( dxr2, "FBgn0010909", displayTranscripts=TRUE, legend=TRUE,cex.axis=1.2, cex=1.3, lwd=2 )
#
#plotDEXSeq( dxr1, "1005", displayTranscripts=TRUE, legend=TRUE,cex.axis=1.2, cex=1.3, lwd=2 )
#
## to display it with the normalised counts.
#
#
##plotDEXSeq( dxr2, "FBgn0010909", expression=FALSE, norCounts=TRUE,legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
#
#
##plotDEXSeq( dxr2, "FBgn0010909", expression=FALSE, splicing=TRUE,legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
#plotDEXSeq( dxr2, "1005", expression=FALSE, splicing=TRUE,legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
#
## WRITE TO BROWSEABLE HTML USING DEXSeqHTML
#
#
##DEXSeqHTML( dxr2, FDR=0.1, color=c("#FF000080", "#0000FF80") 
#
#DEXSeqHTML( dxr1, FDR=0.001, color=c("#FF000080", "#0000FF80") 
#
#DEXSeqHTML( dxr1, FDR=0.001)
#
#
#














